analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.PatternMatching.cs

```diff
@@ -198,6 +198,23 @@ public void RecursivePattern_SetsNotNull_Nested()
             validator.TagValues("End").Should().HaveCount(2).And.OnlyContain(x => x != null && x.HasConstraint(TestConstraint.First));   // 2x because value has different states
         }
 
+        [TestMethod]
+        public void ResursivePattern_ExistingConstraint_DoesNothing()
```

Comment:
Add more variations with different patterns (parametrized test):
### Type pattern

```cs
object value = arg switch
{
    null => null,
    Exception when Condition => null, // Should not create arg=Null
    _ => Tag(""Arg"", arg)
};
```
### Const pattern

```cs
object value = arg switch
{
    null => null,
    1 when Condition => null, // Should not create arg=Null
    _ => Tag(""Arg"", arg)
};
```
### Discard pattern

```cs
object value = arg switch
{
    null => null,
    _ when Condition => null, // Should not create arg=Null
    var _ when Condition => null, // Should not create arg=Null
    _ => Tag(""Arg"", arg)
};
```

### Not pattern

```cs
object value = arg switch
{
    null => null,
    not 1 when Condition => null, // Should not create arg=Null
    _ => Tag(""Arg"", arg)
};
```